gnomad_qc/v4/annotations/compute_coverage.py

```diff
@@ -116,12 +116,43 @@ def main(args):
                 release_only=True,
                 test=test,
                 filter_partitions=range(2) if test else None,
+                annotate_meta=True,
             )
 
+            if args.stratify_by_ukb_and_platform:
+                meta_expr = vds.variant_data.meta
+                ukb_expr = meta_expr.project_meta.ukb_sample
+                strata = [
+                    {"ukb_strata": hl.if_else(ukb_expr, "ukb", "non_ukb")},
+                    {
+                        "subset": hl.or_missing(~ukb_expr, "non_ukb"),
+                        "platform": meta_expr.platform_inference.qc_platform,
+                    },
+                ]
+            else:
+                strata = None
+
             # Compute coverage stats.
-            coverage_ht = compute_coverage_stats(vds, ref_ht, res.interval_ht.ht())
+            coverage_ht = compute_coverage_stats(
+                vds,
+                ref_ht,
+                interval_ht=res.interval_ht.ht(),
+                strata_expr=strata,
+            )
 
             # Checkpoint Table.
+            if args.stratify_by_ukb_and_platform:
+                coverage_ht = coverage_ht.annotate_globals(
+                    coverage_stats_meta=coverage_ht.coverage_stats_meta.map(
+                        lambda x: hl.dict(
+                            x.items().map(
+                                lambda m: hl.if_else(
+                                    m[0] == "ukb_strata", ("subset", m[1]), m
```

Comment:
Any reason to not name this subset originally?

Reply:
yeah, the way the group annotation code works is that is expects the strata with the same name to have the same values, but my first one has "ukb" and "non_ukb", but the second one only has non_ukb, this was a fast way to get the results needed